src/main/java/ether/TransactionsManager.java

```diff
@@ -0,0 +1,663 @@
+package ether;
+import io.left.rightmesh.id.MeshID;
+import io.left.rightmesh.mesh.MeshManager;
+import io.left.rightmesh.proto.MeshTransaction;
+import io.left.rightmesh.util.EtherUtility;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.concurrent.*;
+import static io.left.rightmesh.mesh.MeshManager.TRANSACTION_RECEIVED;
+
+
+/**
+ * The Transactions manager, responsible to process incoming Ether Transactions from Clients in the Mesh network.
+ */
+public final class TransactionsManager {
+
+    private static final int NUM_OF_THREADS = 100;
+
+    private MeshManager meshManager;
+    private LinkedBlockingQueue<MeshManager.MeshTransactionEvent> transactionsQueue = new LinkedBlockingQueue<>();
```

Comment:
consider using abstract interface BlockingQueue as the type unless there is an explicit dependency on functions not defined in the interface. Same for next line.